jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/identity/LDAPUserGroupCallbackImpl.java

```diff
@@ -17,341 +17,116 @@
- *  <li>ldap.user.id.dn (optional, is user id a DN, instructs the callback to query for user DN before searching for roles, default false)</li>
- *  <li>ldap.search.scope (optional, if not given 'OBJECT_SCOPE' will be used) possible values are: OBJECT_SCOPE, ONELEVEL_SCOPE, SUBTREE_SCOPE</li>
- *  <li>java.naming.factory.initial</li>
- *  <li>java.naming.security.authentication</li>
- *  <li>java.naming.security.protocol</li>
- *  <li>java.naming.provider.url</li>
- *  <li></li>
+ * <li>ldap.bind.user (optional if LDAP server accepts anonymous access)</li>
+ * <li>ldap.bind.pwd (optional if LDAP server accepts anonymous access</li>
+ * <li>ldap.user.ctx (mandatory)</li>
+ * <li>ldap.role.ctx (mandatory)</li>
+ * <li>ldap.user.roles.ctx (optional, if not given ldap.role.ctx will be used)</li>
+ * <li>ldap.user.filter (mandatory)</li>
+ * <li>ldap.role.filter (mandatory)</li>
+ * <li>ldap.user.roles.filter (mandatory)</li>
+ * <li>ldap.user.attr.id (optional, if not given 'uid' will be used)</li>
+ * <li>ldap.role.attr.id (optional, if not given 'cn' will be used)</li>
+ * <li>ldap.user.id.dn (optional, is user id a DN, instructs the callback to query for user DN before searching for roles, default false)</li>
+ * <li>ldap.search.scope (optional, if not given 'ONELEVEL_SCOPE' will be used) possible values are: OBJECT_SCOPE, ONELEVEL_SCOPE, SUBTREE_SCOPE</li>
+ * <li>java.naming.factory.initial</li>
+ * <li>java.naming.security.authentication</li>
+ * <li>java.naming.security.protocol</li>
+ * <li>java.naming.provider.url</li>
+ * <li></li>
  * </ul>
+ * </p>
  */
-public class LDAPUserGroupCallbackImpl extends AbstractUserGroupInfo implements UserGroupCallback {
-    
-    private static final Logger logger = LoggerFactory.getLogger(LDAPUserGroupCallbackImpl.class);
-    
-    protected static final String DEFAULT_PROPERTIES_NAME = "classpath:/jbpm.usergroup.callback.properties";
-    
-    public static final String BIND_USER = "ldap.bind.user";
-    public static final String BIND_PWD = "ldap.bind.pwd";
+public class LDAPUserGroupCallbackImpl extends AbstractLDAPUserGroupInfo implements UserGroupCallback {
+
+    private static final String DEFAULT_PROPERTIES_NAME = "jbpm.usergroup.callback";
+
     public static final String USER_CTX = "ldap.user.ctx";
     public static final String ROLE_CTX = "ldap.role.ctx";
     public static final String USER_ROLES_CTX = "ldap.user.roles.ctx";
     public static final String USER_FILTER = "ldap.user.filter";
     public static final String ROLE_FILTER = "ldap.role.filter";
     public static final String USER_ROLES_FILTER = "ldap.user.roles.filter";
     public static final String USER_ATTR_ID = "ldap.user.attr.id";
-    public static final String ROLE_ATTR_ID = "ldap.roles.attr.id";
-    public static final String IS_USER_ID_DN = "ldap.user.id.dn";
+    public static final String ROLE_ATTR_ID = "ldap.role.attr.id";
+    public static final String USER_ID_DN = "ldap.user.id.dn";
```

Comment:
Thanks for the feedback. I have reverted this change in order not to break the backward compatibility.